templates/common/_base/files/configure-ovs-network.yaml

```diff
@@ -0,0 +1,313 @@
+mode: 0755
+path: "/usr/local/bin/configure-ovs.sh"
+contents:
+  inline: |
+    #!/bin/bash
+    set -eux
+
+    # Configures NICs onto OVS bridge "br-ex"
+    # Configuration is either auto-detected or provided through a config file written to
+    # /etc/sysconfig/network-scripts/ovs-config.yaml
+    # Example Config file:
+    # ---
+    # interfaces:
+    #   - name: br-ex
+    #     type: ovs-bridge
+    #     state: up
+    #     bridge:
+    #       port:
+    #         - name: bond0
+    #           vlan:
+    #             mode: access
+    #             tag: 2
+    #           link-aggregation:
+    #             lacp: active
+    #             mode: balance-slb
+    #             slaves:
+    #               - name: eth1
+    #               - name: eth2
+    #         - name: br-ex
+    #   - name: br-ex
+    #     type: ovs-interface
+    #     mtu: 9000
+    #     mac-address: 52:54:00:E0:64:7D
+    #
+    # This configuration file format is the same as what is used by nmstate. Only configuration options listed
+    # here are supported. The only configuration this file should contain is for configuring br-ex. Other interfaces
+    # are not supported.
+
+    parse_yaml() {
+      local yaml_file=$1
+      local prefix=$2
+      local s
+      local w
+      local fs
+
+      s='[[:space:]]*'
+      w='[a-zA-Z0-9_.-]*'
+      fs="$(echo @|tr @ '\034')"
+
+      (
+        sed -e '/- [^\“]'"[^\']"'.*: /s|\([ ]*\)- \([[:space:]]*\)|\1-\'$'\n''  \1\2|g' |
+
+        sed -ne '/^--/s|--||g; s|\"|\\\"|g; s/[[:space:]]*$//g;' \
+            -e "/#.*[\"\']/!s| #.*||g; /^#/s|#.*||g;" \
+            -e "s|^\($s\)\($w\)$s:$s\"\(.*\)\"$s\$|\1$fs\2$fs\3|p" \
+            -e "s|^\($s\)\($w\)${s}[:-]$s\(.*\)$s\$|\1$fs\2$fs\3|p" |
+
```

Comment:
so, this is all kind of terrifying

Why do we have to support defining bonding and vlans in _this_ script? Can't we just require the admin to set that up via other means (eg, writing NetworkManager config files directly) and then have this script just only do the "create an OVS bridge and move the default interface to it" part? (And then if RHCOS gets support for nmstate or whatever later, then people can start using that feature then.)

Reply:
So you are suggesting the user writes the NM key files themselves and places them into the right folder during bring up, and then what, this script just sees the files there and does nothing and allows NM to do its thing?

Reply:
I've removed the NM state config reading and only left the auto detection bits. For custom network configuration the user will need to provide NM key files via ignition.